[22122] Fix WP keeps type of previous work package

oliverguenther opened the pull request:
https://community.openproject.org/work_packages/22122/activity

Supersedes https://github.com/opf/openproject/pull/3886

furinvader committed:
Clear data after saving a work package

furinvader committed:
Spec resetting of new WP form

TeatroIO commented:
I've prepared a stage to preview changes. <a href='http://furinvader-bugs-22122-create-new-wp-keeps-type.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/' target='_blank'>Open stage</a> or <a href='http://furinvader-bugs-22122-create-new-wp-keeps-type.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com//teatro.log' target='_blank'>view logs</a>.

basiszwo commented on spec/features/work_packages/new_work_package_spec.rb:
```diff
@@ -55,8 +51,34 @@ def save_work_package!(expect_success=true)
 
+  def create_work_package(type)
+    work_packages_page.click_toolbar_button 'Work packages'
+    sleep(0.1)
```
I tried to remove this `sleep` as it should not be there. Removing it fails specs although there is no obvious reason.

The sequence is
- click create button
- wait
- click "task"
- check if subject empty

removing the `wait` step makes `check if subject empty` fail.

oliverguenther committed:
Reduce forced sleep in spec

This removes the forced sleep of the spec and rather checks
for the correct page to have loaded before the next is requested.

oliverguenther commented:
Hey @furinvader , the `sleep` seemed to be necessary, but it isnt. This is actually an edge-case of the loading of the work package split screen.

If you're fast enough, you can reproduce it manually. After saving the work package, quickly click on `+ work package` and select a type. No error is printed to the console, yet I'm getting this weird creation form that appears to show a new work package, but did not trigger `editAll`.

![bildschirmfoto 2015-11-24 um 11 18 27](https://cloud.githubusercontent.com/assets/459462/11363949/320af99e-929d-11e5-8bc7-fed49b101504.png)

furinvader commented:
Can this PR be merged or is there anything left to do?

oliverguenther commented:
The issue I commented above remains. If its possible to fix this, I'd prefer this over the more complicate spec.

furinvader commented:
OK, thanks. I'll have a look as soon as I can.

furinvader committed:
Deactivate WP create button when in WP new route
